modules/cudastereo/include/opencv2/cudastereo.hpp

```diff
@@ -241,6 +241,53 @@ class CV_EXPORTS_W StereoConstantSpaceBP : public cuda::StereoBeliefPropagation
 CV_EXPORTS_W Ptr<cuda::StereoConstantSpaceBP>
     createStereoConstantSpaceBP(int ndisp = 128, int iters = 8, int levels = 4, int nr_plane = 4, int msg_type = CV_32F);
 
+/////////////////////////////////////////
+// StereoSGM
+
+/** @brief The class implements the modified H. Hirschmuller algorithm @cite HH08.
+Limitation and difference are as follows:
```

Comment:
The `3x3 median` and `quadratic interpolation` are used in original paper HH08.
So, I don't think these are differences nor limitations.

Reply:
Noted with thanks.

Reply:
> The algorithm matches individual pixels only.

This sentence also might cause misunderstanding, because there actually used 9x7 census.

Reply:
> The 3x3 median and quadratic interpolation are used in original paper HH08.
So, I don't think these are differences nor limitations.

I re-read the paper, I decided removing these description.

> > The algorithm matches individual pixels only.
>
> This sentence also might cause misunderstanding, because there actually used 9x7 census.

I removed this description because following CSCT description shows detail and that you mentioned is right.
Also, `cv::cuda::StereoSGM::getBlockSize` has been changed to return `-1` instead of  `1`.

`blockSize` of `cv::StereoSGBM` seems to mean SAD Window size (from variable name and used location, but I couldn't understand implementation completely...).
Thus, `cv::cuda::StereoSGM::getBlockSize` might had better return `9x7` if possible and doc comment might had better be the following.

```diff
diff --git a/modules/cudastereo/include/opencv2/cudastereo.hpp b/modules/cudastereo/include/opencv2/cudastereo.hpp
index 4cb2cbd1..e93aca57 100644
--- a/modules/cudastereo/include/opencv2/cudastereo.hpp
+++ b/modules/cudastereo/include/opencv2/cudastereo.hpp
@@ -250,7 +250,7 @@ Limitation and difference are as follows:
 -   Cost Aggregation use StereoSGM::MODE_HH or StereoSGM::MODE_HH4 only.
 -   Mutual Information cost function is not implemented.
 Instead, Center-Symmetric Census Transform with \f$9 \times 7\f$ window size from @cite Spangenberg2013
-is used for robustness.
+is used for robustness. Though, The algorithm matches blocks, not individual pixels.

 @sa cv::StereoSGBM
 */
```

@atakagi-fixstars What do you think about it?

Reply:
Tried fixing.

Reply:
Good job.